File: das_backend/src/main/java/ch/sbb/backend/admin/infrastructure/configuration/ServicePointsBeanConfiguration.java
package ch.sbb.backend.admin.infrastructure.configuration;

import ch.sbb.backend.admin.domain.repository.ServicePointRepository;
import ch.sbb.backend.admin.domain.service.DomainServicePointService;
import ch.sbb.backend.admin.domain.service.ServicePointService;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class ServicePointsBeanConfiguration {

    @Bean
    ServicePointService servicePointService(ServicePointRepository servicePointRepository) {
        return new DomainServicePointService(servicePointRepository);
    }
}

File: backend/src/main/kotlin/ch/sbb/backend/servicepoints/infrastructure/configuration/ServicePointsBeanConfiguration.kt
package ch.sbb.backend.servicepoints.infrastructure.configuration

import ch.sbb.backend.servicepoints.domain.service.DomainServicePointService
import ch.sbb.backend.servicepoints.domain.repository.ServicePointRepository
import ch.sbb.backend.servicepoints.domain.service.ServicePointService
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class ServicePointsBeanConfiguration {

    @Bean
    fun servicePointService(servicePointRepository: ServicePointRepository): ServicePointService {
        return DomainServicePointService(servicePointRepository)
    }

}
